packages/api/test/scripts/test-context.js

```diff
@@ -0,0 +1,118 @@
+import { registerSharedWorker } from 'ava/plugin'
+import { Miniflare } from 'miniflare'
+import path from 'path'
+import { fileURLToPath } from 'url'
+import { serviceConfigFromVariables } from '../../src/config.js'
+const __dirname = fileURLToPath(new URL('.', import.meta.url))
+
+/**
+ *
+ * @param {Record<string, string>} bindings
+ * @returns
+ */
+export function makeMiniflare(bindings = {}) {
+  // Create a new Miniflare environment for each test
+  const envPath = path.join(__dirname, 'test.env')
+  return new Miniflare({
+    // Autoload configuration from `.env`, `package.json` and `wrangler.toml`
+    envPath,
+    packagePath: true,
+    wranglerConfigPath: true,
+    // We don't want to rebuild our worker for each test, we're already doing
+    // it once before we run all tests in package.json, so disable it here.
+    // This will override the option in wrangler.toml.
+    buildCommand: undefined,
+    bindings,
+  })
+}
+
+/**
+ * @param {Record<string, string>} vars
+ */
+export const defineGlobals = (vars) => {
+  /** @type Record<string, unknown> */
+  const globals = globalThis
+  for (const [k, v] of Object.entries(vars)) {
+    globals[k] = v
+  }
+}
+
+/**
+ *
+ * @param {import('ava').ExecutionContext<unknown>} t
+ * @param {object} opts
+ * @param {boolean} [opts.bindGlobals]
+ * @param {boolean} [opts.noContainers]
+ * @param {Record<string, string>} [opts.overrides]
+ */
+export async function setupMiniflareContext(
+  t,
+  { bindGlobals = false, noContainers = false, overrides = {} } = {}
+) {
+  t.timeout(600 * 1000, 'timed out pulling / starting test containers')
+
+  if (!noContainers) {
+    const sharedWorker = await registerSharedWorker({
+      filename: path.resolve(__dirname, 'containers.js'),
+      supportedProtocols: ['ava-4'],
+    })
+
+    // wait for the shared worker to publish an object with env var overrides
+    for await (const message of sharedWorker.subscribe()) {
+      // console.log('got message from ava shared worker', message)
+      if (!message.data || typeof message.data !== 'object') {
+        continue
+      }
+      if (!('overrides' in message.data)) {
+        continue
+      }
+      // console.log('setting environment overrides', message.data)
+      // @ts-ignore
+      overrides = { ...overrides, ...message.data.overrides }
+      break
+    }
+  }
+
+  const mf = makeMiniflare(overrides)
+
+  const bindings = await mf.getBindings()
+  // console.log('miniflare bindings', bindings)
+  const serviceConfig = serviceConfigFromVariables(bindings)
+  if (bindGlobals) {
+    // optionally pull cloudflare bindings into the global scope of the test runner
+    defineGlobals(bindings)
+  }
```

Comment:
why do we need this ?

Reply:
I'm using it to test the config loading code in config.spec.js, so we can test the `loadConfigVariables` function outside of the worker context. I'll take another look at those tests and see if there's another way to do it.

Reply:
I rewrote the config tests to avoid the `bindGlobals` option - the tests now set the global variables directly as part of the test setup.